crypto/include/internal/ec_int.h

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2018 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+/* Internal EC functions for other submodules: not for application use */
+
+#ifndef HEADER_OSSL_EC_INTERNAL_H
+# define HEADER_OSSL_EC_INTERNAL_H
+# include <openssl/opensslconf.h>
+
+# ifndef OPENSSL_NO_EC
+
+#  ifdef __cplusplus
+extern "C" {
+#  endif
```

Comment:
This is internal header and it's never exposed to C++ compiler, hence there is no need for this. But if one chooses to have it, wouldn't after #include <openssl/ec.h> be the right spot?

Reply:
> This is internal header and it's never exposed to C++ compiler, hence there is no need for this.

That's a very good point, and one I wondered about myself: I went for inclusion when I copied the basic structure form the recent `crypto/include/internal/sm2.h`.

My final rationale in keeping also what didn't make much sense to me was that this header was very recently added after a long review, and that reviewer definitely knew better!

Reply:
> My final rationale in keeping also what didn't make much sense to me was that this header was very recently added after a long review, and that the reviewers definitely knew better!

No, the reviewers just missed it! :-(

Reply:
@dot-asm I put #6554 to fix also the SM2 internal header

Reply:
@mattcaswell It's not like reviewers are reviewing tons of code these days!! 😛 

BTW, I'm trying out the `git commit --fixup` trick you suggested: thanks, that's very useful!!

Reply:
That specific header was not added, but moved from public. Well, it's not really an excuse, just an explanation...